src/tasks/select-resource.ts

```diff
@@ -0,0 +1,108 @@
+import inquirer from "inquirer";
+import inquirerSearchList from "inquirer-search-list";
+import { Task, TaskStep, pauseSpinningAndExecuteTask, transitionTo } from "../util/task";
+import { ZeplinResource, ResourceContext } from "./context/resource";
+import * as ui from "./ui/select-resource";
+import { ZeplinApi } from "../api";
+import logger from "../util/logger";
+import { CLIError } from "../errors";
+
+inquirer.registerPrompt("search-list", inquirerSearchList);
+
+const zeplinApi = new ZeplinApi();
+
+/**
+ * Workaround for inquirer-search-list to print value name
+ * on terminal but retrieve the actual value of the user's choice
+ */
+class Choice<T extends { name: string }> {
+    value: T;
+    constructor(choice: T) {
+        this.value = choice;
+    }
+
+    toString(): string {
+        return this.value.name;
+    }
+}
+
+function createChoice(resource: ZeplinResource): { name: string; value: Choice<ZeplinResource> } {
+    let { name } = resource;
+    if (resource.organization) {
+        name = `${resource.name} (${resource.organization.name} org.)`;
+    }
+    return {
+        name,
+        value: new Choice(resource)
+    };
+}
+
+const retrieveResources: TaskStep<ResourceContext> = async (ctx): Promise<void> => {
+    const resources: Record<string, ZeplinResource> = {};
+    try {
+        const { projects } = await zeplinApi.getProjects(ctx.auth.token);
+        projects.forEach(p => {
+            resources[p._id] = {
+                _id: p._id,
+                name: p.name,
+                type: "Project",
+                organization: p.organization
+            };
+        });
+
+        const { styleguides } = await zeplinApi.getStyleguides(ctx.auth.token);
+        styleguides.forEach(s => {
+            resources[s._id] = {
+                _id: s._id,
+                name: s.name,
+                type: "Styleguide",
+                organization: s.organization
+            };
+        });
+
+        ctx.resources = resources;
```

Comment:
We can change the code to fix eslint error or we can add inline eslint-disable comment instead of disabling the rule via `eslintrc` file.
```suggestion
    ctx.resources = ctx.resources || {};
    const { resources } = ctx;
    try {
        const { projects } = await zeplinApi.getProjects(ctx.auth.token);
        projects.forEach(p => {
            resources[p._id] = {
                _id: p._id,
                name: p.name,
                type: "Project",
                organization: p.organization
            };
        });

        const { styleguides } = await zeplinApi.getStyleguides(ctx.auth.token);
        styleguides.forEach(s => {
            resources[s._id] = {
                _id: s._id,
                name: s.name,
                type: "Styleguide",
                organization: s.organization
            };
        });
```

Reply:
I've specifically ignored the error for tasks, not just for this case. Eslint is warning us based on the object signature. For example most tasks will read `ctx.a` and then update `ctx.b`, this is essential their design. Eslint thinks that a and b could be related and we could have changed something in ctx that would affect the value of `ctx.b`. This is just false positive for us.